apollo-router/src/configuration/mod.rs

```diff
@@ -502,6 +550,15 @@ impl Configuration {
             }
         }
 
+        if self.experimental_query_planner_mode == QueryPlannerMode::New
+            && self.experimental_apollo_metrics_generation_mode != ApolloMetricsGenerationMode::New
+        {
+            return Err(ConfigurationError::InvalidConfiguration {
+                message: "`experimental_query_planner_mode: new` requires `experimental_apollo_metrics_generation_mode: new`",
+                error: "either change to some other query planner mode, or change to new metrics generation".into()
```

Comment:
oooo `experimental_apollo_metrics_generation_mode` will probably not be out of "experimental" by June 17th, eh @BrynCooke? I added this as a note to keep an eye out when writing docs for our June 17th release.